crates/trigger/src/stdio.rs

```diff
@@ -126,3 +151,11 @@ impl std::io::Write for ComponentStdioWriter {
         Ok(())
     }
 }
+
+fn bullet_list<S: std::fmt::Display>(items: impl IntoIterator<Item = S>) -> String {
+    items
+        .into_iter()
+        .map(|item| format!("  - {item}"))
+        .collect::<Vec<_>>()
+        .join("\n")
```

Comment:
_Almost_ equivalent (but with extra trailing \n):

```suggestion
        .map(|item| format!("  - {item}\n"))
        .collect()
```

Reply:
That's nice, but because I return the error message rather than fully controlling the printing it means, I end up with a blank line between the error and the exit code message:

```
$ spin up --follow godby --follow helo
Error: The following component(s) specified in --follow do not exist in the application:
  - helo
  - godby
The following components exist:
  - gogogo
  - hello
  - goodbyte

Error: exit status: 1
```

...which is neither here nor there, I'm sure, but I slightly prefer the more compact form.  If we can get rid of the exit status message then this would be a good change.  (And thanks for mentioning it, I didn't know collecting to String did this.  Though now I do I'm not sure how I feel about it!)